etcdserver/v3_server.go

```diff
@@ -73,7 +73,7 @@ type Authenticator interface {
 }
 
 func (s *EtcdServer) Range(ctx context.Context, r *pb.RangeRequest) (*pb.RangeResponse, error) {
-	if r.Serializable {
+	if r.Serializable && s.AuthStore().IsRangePermitted(r.Header, string(r.Range.Key), string(r.RangeEnd)) {
```

Comment:
RequestHeader isn't prepared in this scope. It should be created from metadata associated with context. See https://github.com/coreos/etcd/blob/master/etcdserver/v3_server.go#L420